src/main/java/org/dependencytrack/resources/v1/AnalysisResource.java

```diff
@@ -170,6 +170,21 @@ public Response updateAnalysis(AnalysisRequest request) {
                 return Response.status(Response.Status.NOT_FOUND).entity("The vulnerability could not be found.").build();
             }
 
+            List<Component> matchingComponents = new ArrayList<>();
+            if(request.isSuppressAllFlag()) {
+                List<Project> allProjects = qm.getProjects().getList(Project.class);
+                for (Project project1 : allProjects) {
+                    List<Component> components = qm.getAllComponents(project1);
+                    for (Component c : components) {
+                        if (c.getName().equals(component.getName()) && c.getVersion().compareTo(component.getVersion()) <= 0) {
+                            matchingComponents.add(c);
+                        }
+                    }
+                }
+            } else {
+                matchingComponents.add(component);
+            }
+
             String commenter = null;
```

Comment:
This looks inefficient is it will retrieve all components one-by-one. There also seems to be an assumption that the suppression needs to happen for all versions older than or equal to the version of the component selected for the suppression?